sunflower-land/sunflower-land
# [FEAT] Revert transactions

# Description

A more seamless session management and transaction flow. Now supports:

1. Non-blocking transactions. You can keep playing while a transaction is in progress without fear of rollbacks
2. Instant transaction retries. Persists the latest signature to allow players to instantly retry
3. Can sync while you have an active auction bid
4. Can buy resources off P2P market from someone with in-flight transaction.

<img width="562" alt="Screenshot 2024-09-10 at 8 52 56 AM" src="https://github.com/user-attachments/assets/f7f5a57a-e904-4e36-a764-a0711a28c856">
<img width="537" alt="Screenshot 2024-09-10 at 8 53 12 AM" src="https://github.com/user-attachments/assets/fe5d9d24-8ace-4b8e-a18e-5021a6a2d199">
<img width="253" alt="Screenshot 2024-09-10 at 8 53 17 AM" src="https://github.com/user-attachments/assets/8ac3cc44-85ff-47cd-b47e-06e7879b692f">
<img width="558" alt="Screenshot 2024-09-10 at 8 53 23 AM" src="https://github.com/user-attachments/assets/87b3ef22-0f36-4813-9b31-ebea23b4bc07">


## How to test?

We will want to test the following flows:
- Store on chain
- Withdrawing (all items)
- Mint Auction item
- Refreshing on chain

1. Point at local API
2. Fire transactions and ensure flow is correct after transaction finishes
3. Start transaction and wait for timeout (revert should occur)
